provision/acc_provision/acc_provision.py

```diff
@@ -2073,6 +2073,39 @@ def is_support_dualstack(flavor):
     return False
 
 
+def has_contains_v6(subnet_data):
+    subnet_info = []
+    if not isinstance(subnet_data, list):
+        subnet_info[-1] = subnet_data
+    else:
+        subnets = []
+        for subnet in subnet_data:
+            subnets.append(subnet)
+        for subnet in subnets:
+            subnet_info.append(subnet)
+
+    for subnet in subnet_info:
+        rtr, _ = subnet.split("/")
+        ip = ipaddress.ip_address(rtr)
+        if ip.version == 6:
+            return True
+
+    return False
+
+
+def is_valid_dualstack_config(config):
+    if "net_config" not in config:
+        return False
+    if not has_contains_v6(config["net_config"]["node_subnet"]) or not has_contains_v6(config["net_config"]["pod_subnet"]):
+        return False
+
+    if has_contains_v6(config["net_config"]["extern_dynamic"]) or has_contains_v6(config["net_config"]["extern_static"]):
+        if has_contains_v6(config["net_config"]["pod_subnet"]) and has_contains_v6(config["net_config"]["node_subnet"]):
+            return True
```

Comment:
We can get/fill this info once and use, no need call func has_contains_v6 multiple time, It would be better if we can have function which check IPV4 Only case, IPV6 Only case and dual stack case, for all other case we can just raise error with appropriate message.

Reply:
addressed